src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitCommitMessageDecXtract.java

```diff
@@ -0,0 +1,172 @@
+package de.uhd.ifi.se.decision.management.jira.extraction.versioncontrol;
+
+import de.uhd.ifi.se.decision.management.jira.model.DecisionKnowledgeElement;
+import de.uhd.ifi.se.decision.management.jira.model.DocumentationLocation;
+import de.uhd.ifi.se.decision.management.jira.model.KnowledgeType;
+import de.uhd.ifi.se.decision.management.jira.model.impl.DecisionKnowledgeElementImpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * purpose: extract decision knowledge elements from single git
+ * commit fullMessage text.
+ *
+ *
+ * Decision knowledge can be documented in commit messages
+ * using following syntax:
+ * """
+ * [decKnowledgeTag]knowledge summary text
+
+ * knowledge description text after empty[/decKnowledgeTag]
+ * """
+ *
+ * where [decKnowledgeTag] belongs to set of know Knowledge Types,
+ * for example issue, alternative, decision etc.
+ */
+public class GitCommitMessageDecXtract {
```

Comment:
@lw2011, great that you inserted this class! This is a fancy name, which I like, but it might be misleading for people who are not familiar with the ConDec plugin. In the Eclipse plugin, I called it CommitMessageParser:
https://github.com/cures-hub/cures-condec-eclipse/blob/master/de.uhd.ifi.se.decision.management.eclipse/src/de/uhd/ifi/se/decision/management/eclipse/extraction/CommitMessageParser.java